src/borg/_hashindex.c

```diff
@@ -39,6 +40,18 @@ typedef struct {
     int32_t num_buckets;
     int8_t  key_size;
     int8_t  value_size;
+}) HashHeader1;
+
+BORG_PACKED(
+typedef struct {
+    char magic[MAGIC_LEN];
+    int32_t version;
+    int32_t num_entries;
+    int32_t num_buckets;
+    int32_t num_empty;
+    int32_t key_size;
+    int32_t value_size;
+    char reserved[1024 - 32];  // filler to 1024 bytes total
```

Comment:
Do we want to go for 4096, a common memory page size, right away?

I don't see any obvious benefit to it right now, but I wanted to raise the question anyway in case someone else has an idea.

Reply:
I thought about this and guess it is pointless:

The size of the hashtable is header_size + N * bucket_size and N is pretty arbitrary (currently guess it is usually a prime value due to some unneccessary change in the past).

So, no matter what we choose for header_size, it will use a fraction of a page at the end.
Due to that, it does not matter if we use a fraction of a page at the beginning also.

Besides that, page size on apple silicon (e.g. M1) is 16k.

So I just picked 1k because:
- not too big so it would add significant overhead
- not too small to ever would need to get changed

Reply:
Another thing is the on-disk block size for HDDs and SSDs (in case we ever use mmap again), but guess the same consideration applies there also.

Reply:
TIL about a relevant architecture with non-4k pages. So I guess it indeed does not matter at all (and I don't think there are actual cases where one might want to map the rest of the hashindex, but not the header).